File: testdata/jennies/rawtypes/struct_with_complex_fields/PHPRawTypes/src/StructComplexFields/SomeOtherStruct.php
<?php

namespace Grafana\Foundation\StructComplexFields;

class SomeOtherStruct implements \JsonSerializable
{
    /**
     * @var mixed
     */
    public $fieldAny;

    /**
     * @param mixed|null $fieldAny
     */
    public function __construct( $fieldAny = null)
    {
        $this->fieldAny = $fieldAny ?: null;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{FieldAny?: mixed} $inputData */
        $data = $inputData;
        return new self(
            fieldAny: $data["FieldAny"] ?? null,
        );
    }

    /**
     * @return array<string, mixed>
     */
    public function jsonSerialize(): array
    {
        $data = [
            "FieldAny" => $this->fieldAny,
        ];
        return $data;
    }
}

File: testdata/jennies/rawtypes/struct_with_complex_fields/CSharpRawTypes/src/Grafana/Foundation/StructComplexFields/SomeOtherStruct.cs
namespace Grafana.Foundation.StructComplexFields;


public class SomeOtherStruct
{
    public object FieldAny;

    public SomeOtherStruct()
    {
        this.FieldAny = new object();
    }

    public SomeOtherStruct(object fieldAny)
    {
        this.FieldAny = fieldAny;
    }
}
